src/test/java/com/list/todo/it/TodoListTest.java

```diff
@@ -1,221 +1,589 @@
+    private static final String TODO_LIST_COMMENT = "comment";
+
+    private HandlerMethodArgumentResolver putAuthenticationPrincipal = new HandlerMethodArgumentResolver() {
+        @Override
+        public boolean supportsParameter(MethodParameter parameter) {
+            return parameter.getParameterType().isAssignableFrom(UserPrincipal.class);
+        }
+
+        @Override
+        public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
+                                      NativeWebRequest webRequest, WebDataBinderFactory binderFactory) throws Exception {
+            UserPrincipal userPrincipal = new UserPrincipal();
+            userPrincipal.setId(USER_ID);
+            userPrincipal.setUsername(USER_USERNAME);
+            userPrincipal.setPassword(new BCryptPasswordEncoder().encode("root"));
+
+            return userPrincipal;
+        }
+    };
+
+    private HandlerMethodArgumentResolver putPageable = new HandlerMethodArgumentResolver() {
+
+        @Override
+        public boolean supportsParameter(
+                MethodParameter parameter) {
+            if (parameter.getParameterType().equals(
+                    Pageable.class)) {
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public Object resolveArgument(
+                MethodParameter parameter,
+                ModelAndViewContainer mavContainer,
+                NativeWebRequest webRequest,
+                WebDataBinderFactory binderFactory) throws Exception {
+
+            return new PageRequest(0, 50);
+        }
+    };
+
+    @Before
+    public void before() {
+        mockMvc = MockMvcBuilders
+                .standaloneSetup(todoListController)
+                .setCustomArgumentResolvers(putAuthenticationPrincipal, putPageable)
+                .build();
+    }
 
     @Test
-    public void getTodoListsByUser() throws Exception {
-        MockHttpServletResponse response = this.mockMvc.perform(get("/api/todolists/my"))
+    public void getTodoLists_getExistentTodoLists_OkStatus() throws Exception {
+        List<TodoList> todoLists = ObjectsProvider.createListOfTodoLists();
+
+        when(todoListService.getTodoListsByUser(any(UserPrincipal.class), eq(TodoListStatus.ACTIVE), any(Pageable.class), eq(new ArrayList<>())))
+                .thenReturn(todoLists);
+
+        MvcResult result = this.mockMvc.perform(get("/api/todolists?status={status}&tagId=", "ACTIVE")
+                .with(user(USER_USERNAME))
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
                 .andDo(print())
-                .andExpect(authenticated())
+                .andExpect(status().isOk())
+                .andReturn();
+
+        List<TodoList> todoListsFromJsonResponse = jsonParser.getListOfObjectsFromJsonResponse(result.getResponse().getContentAsString());
+
+        todoLists.sort(idComparator);
```

Comment:
create own method, something like assertListsEqual() and do sorting and actual assertion inside it
Remove idComparator field, as you need it only once in that method